File: html/cgi/php/LogOut.php
<?php
// Enable error reporting for debugging
error_reporting(E_ALL);
ini_set('display_errors', 1);

// Clear cookies by setting their expiration date to a past date
setcookie('user', '', time() - 3600, '/');
setcookie('bgcolor', '', time() - 3600, '/');

// Build the HTML body
$body = "<!DOCTYPE html>
<html>
<head>
    <title>Bye Bye!</title>
    <style>
        * { font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif; }
        body { background-color: CornflowerBlue; display: flex; justify-content: center; align-items: center; height: 100vh; margin: 0; flex-direction: column; }
        h1 { font-size: 3em; color: black; }
    </style>
</head>
<body>
    <h1>See Ya!</h1>
    <br><a href=\"/php/welcome.php\"><button>Go back to main page</button></a>
</body>
</html>";

// Print the HTTP headers
header("Content-Type: text/html");
header("Content-Length: " . strlen($body));

// Print the HTML body
echo $body;

// Print server name and date if available
if (isset($_SERVER['SERVER_NAME'])) {
    $serverName = $_SERVER['SERVER_NAME'];
    echo "Server: $serverName<br>";
}

$date = gmdate("D, d M Y H:i:s T");
echo "Date: $date<br>";
?>

File: html/cgi/perl/LogOut.pl
#!/usr/bin/perl
use strict;
use warnings;
use POSIX qw(strftime);

# Define the logOut subroutine
sub logOut {
    # Initialize HTML body
    my $body = "<!DOCTYPE html>";
    $body .= "<html><head><title>Bye Bye!</title>";
    $body .= "<style>* {font-family: system-ui, -apple-system, BlinkMacSystemFont,";
    $body .= "'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;}";
    $body .= "body { background-color: CornflowerBlue; display: flex; justify-content: center;";
    $body .= "align-items: center; height: 100vh; margin: 0; flex-direction: column;} ";
    $body .= "h1 { font-size: 3em; color: black; } </style></head>";
    $body .= "<body><h1>See Ya!</h1>";
    $body .= "<br><a href=\"/perl/welcome.pl\"><button>Go back to main page</button></a></body></html>";

    # Print server and date info
    if (exists $ENV{'SERVER_NAME'}) {
        my $server_name = $ENV{'SERVER_NAME'};
        print "Server: $server_name\n";
    }

    my $datetimenow = strftime "%a, %d %b %Y %H:%M:%S %Z", localtime;
    print "Date: $datetimenow\r\n";

    my $expires_time = "Thu, 01 Jan 1970 00:00:10 GMT";
    print "Set-Cookie: user=blbkvvv; Expires=$expires_time;\n";
    print "Set-Cookie: bgcolor=spspspsp; Expires=$expires_time;\n";
    print "Set-Cookie: perls=0; Expires=$expires_time;\n";
    # Print the HTTP headers and body content
    print "Content-Type: text/html\n";
    print "Content-Length: " . length($body) . "\r\n\r\n";
    print "$body";
}

# Call the logOut subroutine if this script is executed directly
logOut() if __FILE__ eq $0;
